### ISSUE TITLE
[java] Make SingularField consider nested classes by default

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3629]
**Affects PMD Version:6.39.0** 

**Rule:SingularField**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_design.html#singularfield>

**Description:**
PMD should report a warning at line 3, but it doesn't. Actually, it is a SingualrField `a`. So, I think it is a false negative. Thanks for your consideration and I am happy to help fix this issue.

**Code Sample demonstrating the issue:**

```java
class Tester {
  class nestedClass {
      private int a;
      public int func(int b) {
          a = b + 5;
          return a;
      }
  }
}
```

**Expected outcome: A warning**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by adangel** [comment]
Thanks for the report. Yes, this is one case that the rule should detect.
The rule has a property called "checkInnerClasses" - if you set this to "true", this violation will be found. The reason, why we have this property is here: https://github.com/pmd/pmd/blob/f18ceebd7bac962ee431a2e7ab3a3458f06b624c/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/SingularFieldRule.java#L146-L147

It would be great, if the rule would do the "right thing" under all circumstances without the need of such a property. Feel free to give it a try.

**Comment by Zustin** [comment]
Hi @adangel Andreas,thanks for your reply and help. I also submitted some other issues (#3628 #3632 #3633 ) and luckily they are similar to the bugs you recently fixed. Could you please help check them? I think they are false negatives and I can provide help for fixing them if you need. Thanks.

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
